ocs/ocs_feed.py

```diff
@@ -226,30 +229,80 @@ def publish_message(self, message, timestamp=None):
             self.agent.publish(self.address, (message, self.encoded()))
 
     @staticmethod
-    def verify_message_data_type(message):
+    def verify_message_data_type(value):
         """Aggregated Feeds can only store certain types of data. Here we check
         that the type of all data contained in a message's 'data' dictionary are
         supported types.
 
         Args:
-            message (dict):
-                Data to be published (see Feed.publish_message for details).
+            value (list, float, int):
+                'data' dictionary value published (see Feed.publish_message for details).
 
         """
         valid_types = (float, int)
 
-        for k, v in message['data'].items():
-            # multi-sample check
-            if isinstance(v, list):
-                if not all(isinstance(x, valid_types) for x in v):
-                    type_set = set([type(x) for x in v])
-                    invalid_types = type_set.difference(valid_types)
-                    raise TypeError("message 'data' block contains invalid data" +
-                                    f"types: {invalid_types}")
+        # multi-sample check
+        if isinstance(value, list):
+            if not all(isinstance(x, valid_types) for x in value):
+                type_set = set([type(x) for x in value])
+                invalid_types = type_set.difference(valid_types)
+                raise TypeError("message 'data' block contains invalid data" +
+                                f"types: {invalid_types}")
 
-            # single sample check
-            else:
-                if not isinstance(v, valid_types):
-                    invalid_type = type(v)
-                    raise TypeError("message 'data' block contains invalid " +
-                                    f"data type: {invalid_type}")
+        # single sample check
+        else:
+            if not isinstance(value, valid_types):
+                invalid_type = type(value)
+                raise TypeError("message 'data' block contains invalid " +
+                                f"data type: {invalid_type}")
+
+    @staticmethod
+    def verify_data_field_string(field):
+        """There are strict rules for the characters allowed in field names.
+        This function verifies the names in a message are valid.
+
+        A valid name:
+
+        * contains only letters (a-z, A-Z; case sensitive), decimal digits (0-9), and the
+          underscore (_).
+        * begins with a letter, or with any number of underscores followed by a letter.
+        * is no more than 255 characters long.
```

Comment:
As before... this passes '' and should not.

Reply:
I don't think this was addressed in yesterday's commits.

Reply:
Yeah, haven't finished addressing everything yet, just wanted to push my latest work, sorry if that gave the impression I was finished up. Can re-request review when that's true.

Reply:
This should be resolved (and tested) by c46f95c.